src/strategy/dungeons/wotlk/forgeofsouls/FosTriggers.cpp

```diff
@@ -0,0 +1,44 @@
+#include "FosTriggers.h"
+#include "Playerbots.h"
+#include "AiObject.h"
+#include "AiObjectContext.h"
+
+bool MoveFromBronjahmTrigger::IsActive()
+{
+    Unit* boss = nullptr;
+    boss = AI_VALUE2(Unit*, "find target", "bronjahm");
+    if (boss && boss->HasUnitState(UNIT_STATE_CASTING))
+    {
+        if (boss->FindCurrentSpellBySpellId(SPELL_CORRUPT_SOUL))
```

Comment:
Can be combined with the above check as there's no further logic forks needed, no need to add an extra depth here. We don't really even need to check both conditions, the spellId should be enough on it's own, but it's fine to check both